plugins/sip/lib/model/providers/pexip/kPexipUtils.php

```diff
@@ -80,6 +81,54 @@ public static function validateAndRetrieveEntry($entryId)
 		return $dbEntry;
 	}
 
+
+	/**
+	 * @param $sourceType
+	 * @param LiveStreamEntry $dbLiveEntry
+	 * @return entry|LiveStreamEntry|null
+	 * @throws PropelException
+	 */
+	public static function validateAndRetrieveDualStreamEntry(LiveStreamEntry $dbLiveEntry, $sourceType = KalturaSipSourceType::PICTURE_IN_PICTURE)
+	{
+		$dualStreamLiveEntry = null;
+		if ($sourceType != KalturaSipSourceType::PICTURE_IN_PICTURE)
+		{
+			$dualStreamLiveEntryId = $dbLiveEntry->getSipDualStreamEntryId();
+			if ($dualStreamLiveEntryId)
+			{
+				$dualStreamLiveEntry = entryPeer::retrieveByPK($dualStreamLiveEntryId);
+			}
+			if (!$dualStreamLiveEntry)
+			{
+				$liveEntryService = new LiveStreamService();
+				$dualStreamLiveEntry = $liveEntryService->duplicateTemplateEntry($dbLiveEntry->getConversionProfileId(), $dbLiveEntry->getId(), new LiveStreamEntry());
+				$liveEntryService->setBroadcastinUrlsAndStreamPassword($dualStreamLiveEntry);
+				$dualStreamLiveEntry->setDisplayInSearch(false);
+				$dualStreamLiveEntry->setIsSipEnabled(true);
+				$dualStreamLiveEntry->setParentEntryId($dbLiveEntry->getId());
+				$dualStreamLiveEntry->setStatus(entryStatus::READY);
+				$dualStreamLiveEntry->save();
+				$liveEntryService->setFlavorsAsReady($dualStreamLiveEntry);
+			}
+
+			switch ($sourceType)
+			{
+				case KalturaSipSourceType::TALKING_HEADS:
+				{
+					$dualStreamLiveEntry->setName($dbLiveEntry->getId() . '- Screen Share');
+					break;
+				}
+				case KalturaSipSourceType::SCREEN_SHARE:
+				{
+					$dualStreamLiveEntry->setName($dbLiveEntry->getId() . '- Talking Heads');
+					break;
```

Comment:
Handle the default case

Reply:
the default case is an error throw. we check validity when we start the call.